### PR TITLE
Update Toshiba doc for new Toshiba AC IR protocol

### PR DESCRIPTION
**Pull request by kbx81**
## Description:

Updates Toshiba IR Climate documentation based on https://github.com/esphome/esphome/pull/1987

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook. **N/A**

### CODE REVIEW

File: components/climate/ir_climate.rst [hunk 1]
```diff
@@ -154,11 +139,63 @@ Configuration variables:
         header_high: 3265us # AC Units from LG in Brazil, for example use these timings
         header_low: 9856us
 
+
+.. _toshiba:
+
+``toshiba`` Climate
+---------------------
```
Comment (glmnet) [inline components/climate/ir_climate.rst]: Too many dashes?

File: components/climate/ir_climate.rst [hunk 2]
```diff
@@ -62,8 +62,8 @@ controller unit.
       - platform: coolix       # adjust to match your AC unit!
         name: "Living Room AC"
 
-Configuration variables:
-------------------------
+Configuration Variables
```
Comment (glmnet) [inline components/climate/ir_climate.rst]: By undocumented convention this title ends in : (see all other pages) if you want you can make it not be a title. (But keep the :)

It's actually expected the : for automatic parsing of it.

**Review by glmnet — CHANGES_REQUESTED** [review]
Just some details.

File: components/climate/ir_climate.rst (hunk 1 above)
Comment (kbx81) [reply components/climate/ir_climate.rst]: Too many dashes. I thought the linter catches these...

File: components/climate/ir_climate.rst (hunk 2 above)
Comment (kbx81) [reply components/climate/ir_climate.rst]: Noted! 😄

**Review by glmnet — APPROVED** [review]
Thanks

**Pull request merged**